docs/provisioning-scenarios.md

```diff
@@ -0,0 +1,34 @@
+# ESI Provisioning Scenarios
+
+This document highlights ESI provisioning scenarios after a user has leased a node.
+
+## Policy Pre-Requisites
+
+The policy files for Ironic and Neutron have to be updated as per the [deployment doc](xdocs/deployment.md). These changes allow non-administrative users to access the necessary APIs.
+
+## Metalsmith
+
+[Metalsmith](https://docs.openstack.org/metalsmith/latest/) is a CLI based tool that can be used to provision baremetal nodes using OpenStack services such as Ironic, Neutron, and Glance.
+
+```
+metalsmith deploy --image centos7 --network provisioning --ssh-public-key ~/.ssh/id_rsa.pub --resource-class baremetal
+```
+
+## External Provisioning
+
+1. Create a Neutron network and subnet corresponding to the external provisioning network
+2. Attach the external provisioning network to the node
+3. Power on the node
+
+```
+openstack network create --share --provider-network-type vlan --provider-physical-network datacentre --provider-segment 1234 external-provisioning
+openstack subnet create \
+  --network external-provisioning \
+  --subnet-range 192.168.17.0/24  \
+  --ip-version 4 \
+  --gateway 192.168.17.254 \
+  --allocation-pool start=192.168.17.150,end=192.168.17.159 \
+  --dhcp external-provisioning-subnet
+openstack esi node network attach node-A external-provisioning
+openstack baremetal node power on node-A
```

Comment:
Is there a step here to let Ironic know that a node has been provisioned? I'm thinking back to our conversation with the Ironic team at the PTG.

Reply:
Ah - you can't actually attach a network to a node without moving it to the 'active' state, so 'esi node network attach' does that for you. So in the end, your node will be in an active state.

Reply:
Interesting. So the comments that Julia had re: some sort of dummy provisioning workflow are probably not important after all?

Reply:
At least for this particular workflow, it does not apply.